Add EP25 smart fixture

sdb9696 opened the pull request.

rytilahti reviewed (approved):
The more, the merrier! Could you add the model to the README file, too?

sdb9696 commented:
So I've added the model to the README but while doing so I noticed that https://github.com/python-kasa/python-kasa/pull/567 had tripped up over the partially implemented splitting out of `DEVICES_IOT` vs `DEVICES_SMART` in `conf_test.py` (KP125M added to both `PLUGS` and `PLUGS_SMART`).  This wasn't breaking tests because of the equally confusing default `protocol_filter={"IOT"}` in `filter_model`.

So to get ahead of any future confusion I've updated the PR to:
- Complete the job @rytilahti started of neatly splitting out `conftest.py` `BULBS_IOT` vs `BULBS_SMART` for the other device types.
- Add `protocol_filter` to all the `parameterize` test method calls and made the default behaviour no filter so it's clearer what each filter is doing.
- Updated the `filter_model` function so that it does an exact match on model name instead of fuzzy match (so KP125 doesn't return the KP125M fixture for example).
- Added a section the README listing EP25 and KP125M with SMART. protocol as having limited support like TAPO.
- Added a section the README stating that the API for TAPO/SMART is currently in flux and could change in future releases.

rytilahti reviewed (approved):
LGTM, thanks!
